Fix/333 fix visibility logic in search screen

poula22 opened the pull request:
- search title should appear if there is a history result or network result
- show history result if present
- show network result if present
- search result icon should change based on keyboard shown or not
- for empty state from network show suggested keywords title only
- for empty state of both network and history don't show any thing
- fix icon back click shadow
- fix icon mic click shadow
- fix arrow and close icon shadow
- fix recent section delete animation
- fix shadow of clicked search item
- refactor 

https://github.com/user-attachments/assets/1e59163b-f557-4194-99a6-863713f6e6d3

poula22 committed:
fix: - search title should appear if there is a history result or network result
-show history result if present
- show network result if present
- search result icon should change based on keyboard shown or not
- show suggested keywords title only for network empty state
- fix icon back click shadow
- fix icon mic click shadow
- fix arrow and close icon shadow
- fix recent section delete animation
- fix shadow of clicked search item
- refactor: VoiceSearchHelper

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

poula22 committed:
fix: - fix icon click action in search item

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/components/ExploreHeader.kt:
```diff
@@ -53,39 +51,42 @@ fun ExploreHeader(
-                        .padding(start = 10.dp)
-                        .size(20.dp)
+                        .padding(vertical = 6.dp, horizontal = 4.dp)
+                        .size(40.dp)
```
the previous size and padding were right. why did you change it??

poula22 replied:
same total padding

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchScreen.kt:
```diff
@@ -91,13 +95,19 @@ private fun SearchContent(
                 }
                 interactions.onVoiceSearchFinished()
+            },
+            onSoundLevelChange = { level ->
```
why did you add this empty body?

poula22 replied:
this will be changed. all implementation of recognizer will be replaced with Google dialog

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchScreen.kt:
```diff
@@ -115,7 +125,7 @@ private fun SearchContent(
     val focusRequester = remember { FocusRequester() }
     LaunchedEffect(Unit) {
-        focusRequester.requestFocus()
+        if (state.query.isBlank()) focusRequester.requestFocus()
```
why did you add this condition? we already need this field to be focused once i enter the screen, whether its content is empty or not

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchScreen.kt:
```diff
@@ -176,51 +186,53 @@ private fun LazyListScope.keywordsSection(
-            onClickableText = onClearClick
-        )
-    }
+    val isRecentKeyWordsVisible = recentKeywords.isNotEmpty()
```
many variables make it harder to read

poula22 replied:
I need them to make condition of search title readable

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchScreen.kt:
```diff
@@ -176,51 +186,53 @@ private fun LazyListScope.keywordsSection(
+            SearchItem(
+                text = keyWord,
+                isRecent = true,
+                postfixIcon = if (isKeyboardVisible) painterResource(Theme.icons.outline.arrowRightUp)
```
what is the relation between Keyboard Visibility and the close icon to clear the item??

poula22 replied:
check issue. icon changes based on keyboard visibility from close to arrow

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchViewModel.kt:
```diff
@@ -33,6 +33,7 @@ class SearchViewModel(
 ) : BaseViewModel<SearchScreenState>(SearchScreenState()),
     SearchInteractionListener {
     init {
+        updateState { it.copy(isLoading = true) }
```
you can remove it and just initialize it in the SearchScreenState

poula22 replied:
I forget to remove it. I don't need loading here at all.❤️

skaik-mo commented on designsystem/src/main/java/com/giraffe/designsystem/composable/PosterListSection.kt:
```diff
@@ -21,12 +28,18 @@ fun PosterListSection(
+        visible = posters.isNotEmpty(),
+        enter = fadeIn(fadeAnimationSpec),
+        exit = fadeOut(fadeAnimationSpec) + slideOutHorizontally(slideAnimationSpec) { -it }
+    ) {
```
In RTL, swipe to remove still behaves like LTR. Adjust swipe direction based on layout.

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

poula22 committed:
fix: - make delete support RTL
- remove loading from SearchScreen state
- make search field focusable when has query

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

# Conflicts:
#	presentation/explore/src/main/java/com/giraffe/explore/components/ExploreHeader.kt

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

poula22 committed:
fix: - handle search result issue

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

# Conflicts:
#	designsystem/src/main/res/drawable/outline_arrow_left.xml
#	designsystem/src/main/res/drawable/outline_arrow_right_up.xml
#	presentation/explore/src/main/java/com/giraffe/explore/components/ExploreHeader.kt
#	presentation/explore/src/main/java/com/giraffe/explore/components/SearchItem.kt
#	presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchScreen.kt

poula22 committed:
Merge branch 'develop' into fix/333-fix-visibility-logic-in-search-screen

poula22 committed:
Merge remote-tracking branch 'origin/develop' into fix/333-fix-visibility-logic-in-search-screen

# Conflicts:
#	presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchScreen.kt
#	presentation/explore/src/main/java/com/giraffe/explore/util/VoiceSearchHelper.kt
